### PR TITLE
feat: [SDKCF-3266] add clearing for campaign to display queue

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
add closed counter to the `CampaignData` to consider in the number of events to trigger.

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt [hunk 1]
```diff
@@ -75,12 +74,13 @@ internal class InApp(
...
 
-    override fun closeMessage() {
+    override fun closeMessage(clearQueuedCampaigns: Boolean) {
         val id = displayManager.removeMessage(getRegisteredActivity())
 
-        if (id != null) {
-            // Remove message from ReadyForDisplayMessageRepository.
-            val message = ReadyForDisplayMessageRepository.instance().removeMessage(id as String)
+        if (clearQueuedCampaigns) {
+            ReadyForDisplayMessageRepository.instance().clearMessages(clearQueuedCampaigns)
```
Comment (imaginaris) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt:81]: can this method be called with `false`. Does that even make sense?

Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt:81]: Inside `closeMessage()` API, no. But yes, it is possible to call the method with `false` input.
In that case, the campaigns' `timesClosed` will not be incremented.

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/MessageRepository.kt [hunk 2]
```diff
@@ -24,4 +24,10 @@ internal interface MessageRepository {
      * This is done during session update due to user info update.
      */
     fun clearMessages()
+
+    /**
+     * Clears all message from the repository.
```
[the remark this answers is not in the document]
Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/MessageRepository.kt:29]: Updated. :)

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt (hunk 1 above)
Comment (imaginaris) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt:94]: yeah but is `clearMessages(false)` used anywhere?

Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt:94]: yes, in `SessionManager#onSessionUpdate()`.

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/ReadyForDisplayMessageRepository.kt [hunk 3]
```diff
@@ -30,16 +30,27 @@ internal abstract class ReadyForDisplayMessageRepository : ReadyMessageRepositor
...
+                        false
+                    }
                 }
             }
         }
 
-        override fun clearMessages() {
+        override fun clearMessages(isClearQueued: Boolean) {
             synchronized(messages) {
+                if (isClearQueued) {
```
Comment (imaginaris) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/ReadyForDisplayMessageRepository.kt:51]: `isClearQueued`?, not something like `shouldIncrementTimesClosed`?

Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/ReadyForDisplayMessageRepository.kt]: changed to `shouldIncrementTimesClosed`

**Review by imaginaris — APPROVED** [review]
👍

**Pull request merged**